cirq-core/cirq/_compat_test.py

```diff
@@ -263,6 +263,38 @@ def f_with_badly_deprecated_param(new_count):  # pragma: no cover
         # pylint: enable=unused-variable
 
 
+@duet.sync
+async def test_deprecated_parameter_async_function():
+    @deprecated_parameter(
+        deadline='v1.2',
+        fix='Double it yourself.',
+        func_name='test_func',
+        parameter_desc='double_count',
+        match=lambda args, kwargs: 'double_count' in kwargs,
+        rewrite=lambda args, kwargs: (args, {'new_count': kwargs['double_count'] * 2}),
+    )
+    async def f(new_count):
+        return new_count
+
```

Comment:
The test passes with an unchanged `_compat.py` file.
How about adding -

```suggestion
    assert inspect.iscoroutinefunction(f)
```

Reply:
Sounds good! Thank you.

Reply:
> The test passes with an unchanged _compat.py file.

Does this mean `deprecated_parameters` works for async functions without this change?

Reply:
Before, if you added the deprecated_parameter decorator to an async function, the function was no longer async - meaning it would always run synchronously.

Reply:
Got it, thanks for the clarification!